Mathlib/Tactic/Linter/Style.lean

```diff
@@ -68,4 +69,47 @@ initialize addLinter setOptionLinter
 
 end Style.setOption
 
+/-- The `check_declID` linter: if it emits a warning, then a declID is considered
+non-standard style.  -/
+register_option linter.style.check_declID : Bool := {
+  defValue := false
+  descr := "enable the `setOption` linter"
+}
+
+
+namespace Style.checkDeclID
+
+/-- Checks whether a given identifier name contains "__". -/
+def contains_double_underscore (stx : Syntax) : Bool :=
+  match stx.getSubstring? with
+  | some str => 1 < (str.toString.splitOn "__").length
+  | none => false
+
+/-- The `checkDeclID` linter: if this linter emits a warning, then a declID is considered
+non-standard style. Currently we only check if it contains a double underscore ("__") as a
+substring.
+
+**Why is this bad?** Double underscores in theorem names can be considered non-standard style and
```

Comment:
I like these lines! Can you move this above the the `register_option`, then you can just use `@[inherit_doc] here.